src/main/java/wooteco/subway/line/service/LineService.java

```diff
@@ -0,0 +1,86 @@
+package wooteco.subway.line.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.exception.line.LineDuplicatedNameException;
+import wooteco.subway.exception.line.LineNotFoundException;
+import wooteco.subway.line.Line;
+import wooteco.subway.line.dao.LineDao;
+import wooteco.subway.line.dto.request.LineCreateRequest;
+import wooteco.subway.line.dto.request.LineUpdateRequest;
+import wooteco.subway.line.dto.response.LineResponse;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class LineService {
+    private static final Logger log = LoggerFactory.getLogger(LineService.class);
+
+    private final LineDao lineDao;
+
+    public LineService(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    @Transactional
+    public LineResponse save(LineCreateRequest lineCreateRequest) {
+        validatesNameDuplication(lineCreateRequest);
+        Line line = new Line(lineCreateRequest.getName(), lineCreateRequest.getColor());
+        Line newLine = lineDao.save(line);
+        log.info(newLine.getName() + " 노선 생성 성공");
+        return new LineResponse(newLine.getId(), newLine.getName(), newLine.getColor());
+    }
+
+    private void validatesNameDuplication(LineCreateRequest lineCreateRequest) {
+        lineDao.findByName(lineCreateRequest.getName())
+                .ifPresent(l -> {
+                    throw new LineDuplicatedNameException();
+                });
+    }
+
+    public LineResponse findBy(Long id) {
+        Line newLine = findById(id);
+        log.info(newLine.getName() + "노선 조회 성공");
+        return new LineResponse(newLine.getId(), newLine.getName(), newLine.getColor());
+    }
+
+    private Line findById(Long id) {
+        return lineDao.findById(id)
+                .orElseThrow(LineNotFoundException::new);
+    }
+
+    public List<LineResponse> findAll() {
+        List<Line> lines = lineDao.findAll();
+        log.info("지하철 모든 노선 조회 성공");
+        return lines.stream()
+                .map(line -> new LineResponse(line.getId(), line.getName(), line.getColor()))
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public void update(Long id, LineUpdateRequest lineUpdateRequest) {
+        Line line = findById(id);
+
+        validatesNameDuplicationExceptOriginalName(lineUpdateRequest, line);
+
+        Line updatedLine = line.update(lineUpdateRequest);
+
+        lineDao.update(id, updatedLine);
```

Comment:
```suggestion
        lineDao.update(updatedLine);
```

id를 따로 지정하는게 아니라 객체가 가지고 있을테니 그대로 id를 활용하면 파라미터 갯수를 줄일 수 있겠네요